src/plugins/intel_npu/src/plugin/npuw/weights_bank.cpp

```diff
@@ -155,14 +157,116 @@ bool Bank::is_remote(int64_t uid) const {
     return false;
 }
 
+void Bank::serialize(std::ostream& stream) const {
+    using namespace ov::npuw::s11n;
+
+    LOG_INFO("Serializing weights bank...");
+    LOG_BLOCK();
+
+    std::lock_guard<std::mutex> guard(m_mutex);
+
+    // For now only a singular device is supported
+    // Sanity check
+    NPUW_ASSERT(m_device_banks.size() == 1 && "Bank containing several devices can't be serialized");
```

Comment:
Why?

Reply:
Can we leave it as a separate task? For now serialization only supports 1 device for all submodels

Reply:
Yes, but I'd like to understand the reasoning

Reply:
My code had some limitations before. Done properly now